text/0049-network-ids.md

```diff
@@ -0,0 +1,191 @@
+- Feature Name: `network_ids`
+- Start Date: 2022-07-25
+- MCIP PR: [mobilecoinfoundation/mcips#0049](https://github.com/mobilecoinfoundation/mcips/pull/0049)
+- Tracking issue: None Yet
+
+# Summary
+[summary]: #summary
+
+Each running mobilecoin consensus network will be started with a string, the "network id".
+
+Clients which make requests to consensus MAY pass the expected network id with their request.
+If this doesn't match the server, an error is returned, which includes the server-side network id in the response.
+
+# Motivation
+[motivation]: #motivation
+
+Two important and competing engineering goals for all financial software are:
+
+* Enable Rigorous Testing: Properly test the code before it goes live in a test environment that matches prod as closely as possible.
+* Prevent Mixing of Test and Prod: Make it hard for e.g. users to think they are talking to testnet when they are actually talking to production and vice versa.
+
```

Comment:
This was something that Jason surfaced: in a bunch of infra and related docs, we call it "prod" and all the dns and sentry labels contain the string "prod". But in some other contexts and docu we call it "mainnet".

I personally am fine with either string becoming the network id

Right now, "testnet" usually refers to a specific testnet that mobilecoin foundation is running. 
The other networks that we create locally etc. are referred to as "dev networks". The main difference is, the enclave signing keys for dev nets don't have humans in the loop, they are created automatically on an as needed basis and generally don't live off-line somewhere. The testnet has a real off-line signing process just like mainnet, so that making a testnet release involves all the same steps as making a mainnet release.